libs/auth/react/src/lib/Authenticator/Authenticator.tsx

```diff
@@ -94,7 +94,10 @@ export const Authenticator: FC<Props> = ({ children, autoLogin = true }) => {
       return userManager.signinRedirect({
         state:
           authSettings.switchUserRedirectUrl ??
-          getReturnUrl(locationRef.current, authSettings),
+          getReturnUrl(
+            location != null ? location : locationRef.current,
```

Comment:
Instead of passing a new location object to the getReturnUrl we need to make sure the `locationRef.current` is also updated.

So I would instead update the ref instead of passing a conditional to the getReturnUrl. So something like.

```ts
 const switchUser = useCallback(
    async function switchUser(nationalId?: string, newLocation?: Location) {
      if (newLocation) {
        locationRef.current = newLocation
      }

      const args =
        nationalId !== undefined
          ? {
              login_hint: nationalId,
              /**
               * TODO: remove this.
               * It is currently required to switch delegations, but we'd like
               * the IDS to handle login_required and other potential road
               * blocks. Now OidcSignIn is handling login_required.
               */
              prompt: 'none',
            }
          : {
              prompt: 'select_account',
            }

      dispatch({
        type: ActionType.SWITCH_USER,
      })
      return userManager.signinRedirect({
        state:
          authSettings.switchUserRedirectUrl ??
          getReturnUrl(locationRef.current, authSettings),
        ...args,
      })
      // Nothing more happens here since browser will redirect to IDS.
    },
    [userManager, dispatch, authSettings, locationRef],
  )
```